### PR TITLE
[FEATURE] FeatureFinderCentroided Performance

### PR DESCRIPTION
**Pull request by IceFreez3r**
As mentioned here: #4359, the FeatureFinderCentroided is too slow. 
@cbielow gave me the original file from the issue. 
Times before:
`FeatureFinderCentroided took 02:06:33 h (wall), 02:06:26 h (CPU), 6.18 s (system), 02:06:20 h (user).`
Then I've implemented the idea of using ThresholdMover from the issue and:
Times afterwards:
`FeatureFinderCentroided took 31:11 m (wall), 31:07 m (CPU), 4.42 s (system), 31:03 m (user).`

### CODE REVIEW

**Comment by cbielow** [comment]
while looking at something completely unrelated I remembered that there is an even faster solution:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/FORMAT/OPTIONS/PeakFileOptions.h#L106
Set an Intensity range while loading the mzML. Apply somewhere here:
https://github.com/OpenMS/OpenMS/pull/4652/files#diff-a5a737054268e8beb7abdc43a9a61c43R178

**Comment by timosachsenberg** [comment]
Is there a difference? Maybe the threshold filter also takes care that associated float/string/int dataarrays are filtered

**Comment by jpfeuffer** [comment]
One does not even load the 0 intensity data, so I agree there is an advantage.

**Comment by timosachsenberg** [comment]
Agree but we should check if the data arrays are also correctly filtered e.g. ion mobility

**Comment by timosachsenberg** [comment]
Though for FFC it currently does not use them

**Comment by cbielow** [comment]
metadata is handled correctly AFAIKT.
This would also be a different issue -- this is just using the functionality offered by MzMLFile....

**Comment by timosachsenberg** [comment]
ok so the maximum of the range is: PositionType::maxPositive()
and the minimum std::numeric_limits<PositionType>::epsilon()
I guess

File: src/topp/FeatureFinderCentroided.cpp [hunk 1]
```diff
@@ -178,6 +180,9 @@ class TOPPFeatureFinderCentroided :
     PeakFileOptions options;
     options.setMSLevels(vector<Int>(1, 1));
 
+    using RP_TYPE = DRange<1>::PositionType;
+    options.setIntensityRange({std::numeric_limits<RP_TYPE>::epsilon(), RP_TYPE::maxPositive()});
```
Comment (cbielow) [inline src/topp/FeatureFinderCentroided.cpp:184]: should be `std::numeric_limits<RP_TYPE>::min` (which is 1.175494351e-38F) not `epsilon` (which is 1.192092896e-07F)

Comment (cbielow) [reply src/topp/FeatureFinderCentroided.cpp:184]: note: min is positive. lowest would be -max.

Comment (timosachsenberg) [reply src/topp/FeatureFinderCentroided.cpp:184]: yep right will chance

File: src/topp/FeatureFinderCentroided.cpp [hunk 2]
```diff
@@ -42,6 +42,9 @@
 #include <OpenMS/FORMAT/MzQuantMLFile.h>
 #include <OpenMS/METADATA/MSQuantifications.h>
 #include <OpenMS/SYSTEM/File.h>
+#include <OpenMS/FILTERING/TRANSFORMERS/ThresholdMower.h>
```
Comment (cbielow) [inline src/topp/FeatureFinderCentroided.cpp:45]: this could be removed

Comment (timosachsenberg) [reply src/topp/FeatureFinderCentroided.cpp:45]: k will do in another PR - thanks

Comment (IceFreez3r) [reply src/topp/FeatureFinderCentroided.cpp:45]: Changed that on my fork. I don't know if I need to open another PR or if you can just reopen this one

Comment (timosachsenberg) [reply src/topp/FeatureFinderCentroided.cpp:45]: Thanks but I already pushed it to develop. So I guess its fine

**Pull request merged**